// Copyright 2017 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";

option java_multiple_files = true;
option java_package = "sawtooth.stake.protobuf";

// The actually representation of an ''account'' like in ethereum.
message Stake {
    // A counter for tracking unique states
    int64 nonce = 1;

    // The public key of the owner of this account
    string ownerPubKey = 2;

    // The value of stake stored in this account.
    float value = 3;

    // The block number that this stake may be modified.
    int64 blockNumber = 4;

    // A nickname for the owner
    // string name = 5;
}

message StakeList {
    // In case of collisions, we store a list of public keys.
    map<string, Stake> stakeMap = 1;
}